app/controllers/mentees_controller.rb

```diff
@@ -11,10 +11,10 @@ def index
   end
 
   def add_course
+    course_id = params[:course_id]
     current_mentee = Mentee.find(params[:id])
-    current_mentee.update!(course_id: params[:course_id])
+    current_mentee.update!(course_id: course_id)
+    AssignMentor.assign(current_mentee, course_id)
```

Comment:
I like the `AssignMentor` abstraction! Perhaps we can refactor line 14-16 into a `AssignCourse` class as well?